### PR TITLE
feat(metrics): add default metrics for http requests

### PR DESCRIPTION
**Pull request by caitong93**
**What this PR does / why we need it**:

for https://github.com/caicloud/nirvana/issues/121

**Special notes for your reviewer**:

/cc @your-reviewer

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: examples/instrumentation/main.go [hunk 1]
```diff
@@ -17,22 +17,58 @@ limitations under the License.
 package main
 
 import (
+	"context"
+	"math/rand"
```
Comment (Jimexist) [inline examples/instrumentation/main.go]: use `crypto/rand`

Comment (Jimexist) [reply examples/instrumentation/main.go]: but never mind it doesn't matter

File: examples/instrumentation/main.go [hunk 2]
```diff
@@ -17,22 +17,58 @@ limitations under the License.
 package main
 
 import (
+	"context"
```
Comment (ddysher) [inline examples/instrumentation/main.go]: 👆 remove `instrumentation` binary

File: examples/instrumentation/main.go [hunk 3]
```diff
@@ -17,22 +17,58 @@ limitations under the License.
...
+	"time"
+
 	"github.com/caicloud/nirvana"
+	"github.com/caicloud/nirvana/definition"
 	"github.com/caicloud/nirvana/log"
 	"github.com/caicloud/nirvana/plugins/metrics"
 	"github.com/caicloud/nirvana/plugins/profiling"
 )
 
+// This example shows how metrics and profiling plugin work. And the defaults functionality they provide.
```
Comment (ddysher) [inline examples/instrumentation/main.go]: nit: This example shows how metrics and profiling plugin work, and the defaults functionality they provide.

File: plugins/metrics/metrics.go [hunk 4]
```diff
@@ -46,10 +57,71 @@ func (i *metricsInstaller) Name() string {
 func (i *metricsInstaller) Install(builder service.Builder, cfg *nirvana.Config) error {
 	var err error
 	wapper(cfg, func(c *config) {
-		err = builder.AddDescriptor(definition.SimpleDescriptor(definition.Get, c.path, service.WarpHTTPHandler(promhttp.Handler())))
+		i.requestCounter = prometheus.NewCounterVec(
+			prometheus.CounterOpts{
+				Namespace: c.namespace,
+				Name:      "request_count",
+				Help:      "Counter of apiserver requests broken out for each verb, API resource, client, and HTTP response contentType and code.",
```
Comment (ddysher) [inline plugins/metrics/metrics.go]: nit: s/apiserver/server

**Comment by caitong93** [comment]
@ddysher updated, PTAL

**Comment by ddysher** [comment]
LGTM

@kdada

**Pull request merged**